subsys/mpsl/mpsl_init.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <init.h>
+#include <irq.h>
+#include <kernel.h>
+#include <mpsl.h>
+
+void m_assert_handler(const char *const file, const u32_t line)
+{
+	k_oops();
+}
+
+static int mpsl_lib_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int err = 0;
+	mpsl_clock_lf_cfg_t clock_cfg;
+
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+	clock_cfg.source = MPSL_CLOCK_LF_SRC_RC;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+	clock_cfg.source = MPSL_CLOCK_LF_SRC_XTAL;
+#elif CLOCK_CONTROL_NRF_K32SRC_SYNTH
+	clock_cfg.source = MPSL_CLOCK_LF_SRC_SYNTH;
+#else
+#error "Clock source is not defined"
+#endif
+
+	clock_cfg.rc_ctiv = MPSL_RECOMMENDED_RC_CTIV;
+	clock_cfg.rc_temp_ctiv = MPSL_RECOMMENDED_RC_TEMP_CTIV;
+
+	err = mpsl_init(&clock_cfg, SWI5_IRQn, m_assert_handler);
```

Comment:
Shouldn't this SWI number be configurable in Kconfig?

Reply:
It isn't for the ble_controller. Why do you think it should be configurable?

Reply:
It is nice to make it configurable, but not a necessity at this point of time. The open-source link layer hardcodes this SWI as well: https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/swi.h#L35

Reply:
> It isn't for the ble_controller. Why do you think it should be configurable?

> It is nice to make it configurable, but not a necessity at this point of time. The open-source link layer hardcodes this SWI as well

I just saw a hard-coded number, that's why I asked. Making it configurable, or even only using a macro instead (as this number occurs more than once in this file), wouldn't harm probably, but it can make it easier to change this code in some future, if needed.
That's only a suggestion, I won't insist on changing it now.